In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

In [2]:
DiabetesData = pd.read_csv(r"H:\Intellipaat data science\Mahcine learning intellipaat\Assignments\Case Study\CASE-STUDY-8211-DECISION-TREE-04AUG2021104207\diabetes.csv")

In [3]:
DiabetesData.head()

Pregnancies  Glucose  BloodPressure  SkinThickness  Insulin   BMI  \
0            6      148             72             35        0  33.6   
1            1       85             66             29        0  26.6   
2            8      183             64              0        0  23.3   
3            1       89             66             23       94  28.1   
4            0      137             40             35      168  43.1   

   DiabetesPedigreeFunction  Age  Outcome  
0                     0.627   50        1  
1                     0.351   31        0  
2                     0.672   32        1  
3                     0.167   21        0  
4                     2.288   33        1

In [4]:
x=pd.DataFrame(DiabetesData[['Pregnancies', 'BloodPressure', 'BMI', 'Age']])
y=pd.DataFrame(DiabetesData['Glucose'])
print(x)
print(y)

     Pregnancies  BloodPressure   BMI  Age
0              6             72  33.6   50
1              1             66  26.6   31
2              8             64  23.3   32
3              1             66  28.1   21
4              0             40  43.1   33
..           ...            ...   ...  ...
763           10             76  32.9   63
764            2             70  36.8   27
765            5             72  26.2   30
766            1             60  30.1   47
767            1             70  30.4   23

[768 rows x 4 columns]
     Glucose
0        148
1         85
2        183
3         89
4        137
..       ...
763      101
764      122
765      121
766      126
767       93

[768 rows x 1 columns]


In [5]:
from sklearn.model_selection import train_test_split
x_train, x_test, y_train, y_test = train_test_split(x, y, test_size=0.20)

In [6]:
from sklearn.tree import DecisionTreeRegressor
regressor = DecisionTreeRegressor()
regressor.fit(x_train, y_train)

DecisionTreeRegressor()

In [7]:
y_pred = regressor.predict(x_test)

In [8]:
from sklearn.metrics import mean_squared_error
mse = mean_squared_error(y_pred, y_test)
rmse = np.sqrt(mse)
rmse

42.085353129704735

# Building model with 1 input variable and 1 target variable

In [10]:
DiabetesData = pd.read_csv(r"H:\Intellipaat data science\Mahcine learning intellipaat\Assignments\Case Study\CASE-STUDY-8211-DECISION-TREE-04AUG2021104207\diabetes.csv")


In [11]:
x=pd.DataFrame(DiabetesData['Pregnancies'])
y=pd.DataFrame(DiabetesData['Glucose'])
print(x)
print(y)

     Pregnancies
0              6
1              1
2              8
3              1
4              0
..           ...
763           10
764            2
765            5
766            1
767            1

[768 rows x 1 columns]
     Glucose
0        148
1         85
2        183
3         89
4        137
..       ...
763      101
764      122
765      121
766      126
767       93

[768 rows x 1 columns]


In [12]:
from sklearn.model_selection import train_test_split
x_train, x_test, y_train, y_test = train_test_split(x, y, test_size=0.20)

In [13]:
from sklearn.tree import DecisionTreeRegressor
regressor = DecisionTreeRegressor()
regressor.fit(x_train, y_train)

DecisionTreeRegressor()

In [14]:
y_pred = regressor.predict(x_test)

In [15]:
from sklearn.metrics import mean_squared_error
mse = mean_squared_error(y_pred, y_test)
rmse = np.sqrt(mse)
rmse

33.2521626548038

# The second model has a RMSE value of 33.25 and hence it performs better than the first model .

# END OF ASSIGNMENT